backend/example.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+import asyncio
+import datetime
+
+import websockets
+import shogi
+
+
+# {websocket: (board, bool)}
+# bool is if match made
+websocket_to_match = {}
+
+
+async def producer(board):
+    """Sending moves to update client board state.
+
+    """
+
+    # await 
+    await asyncio.sleep(3.0)
+    return board.kif_str()
+    #yield from test_this()
+
+
+async def consumer(board, message):
+    """Validate move and then update the board in memory which
+    corresponds to the connected user/client sending the move
+    (to this server).
+
+    """
+
+    await asyncio.sleep(1.0)
+    board.push_usi(message)
+
+
+def create_board_or_join(websocket):
```

Comment:
i know this algo is really crappy.  a better algo would be to have a list of tuples and then an index which represents match id, then to have to sets "open matches" and "closed matches," ie., the games which require another player, and those which already have both players.